### ISSUE TITLE
Loopback not utilizing connection pool fully

### ISSUE THREAD

**Issue by vishnu1545** [strongloop/loopback-next#7603]
Using loopback datasources for postgres, we provided max connection pool of 20. As soon as app starts it makes two connection. When doing load test using jmeter, it is seen that only those two are being used. But after some time new connection were formed, to match the max number. But in any given second active connection were never more than 3. Also it was like utilizing a stack of connections because it is noticed that the last released connection was used. So, system was never able to use near 10-15 connections although they were there in idle state. This resulted in a tps of 160, for a process which comprised :

http in -> two db insert -> http out  
The model had 200 fields, there are minimum validations, with no hooks, beforeSave, afterSave etc.

Attaching connection logs:

Start: \<timestamp\>  => represents http in
Stop: \<timestamp\> => represents http out

DB-Write Start: \<unique identifie>\<timestamp\> => represents beginning of write process, handing it off to loopback
DB-Write Stop: \<unique identifier\>\<timestamp\> => represents end of write process,

\#\#\# 2021-07-01T13:26:39.589Z Acquired \<connectionId\> => represents acquiring process by pg, got using pg.pool.on('aquire',...)
releasing connection\#\#\# \<connectionId\> => releasing of said connection back to pool
\#\#\# 2021-07-01T13:26:40.148Z Connect \<connectionId\> => Making new connections

Thanks.
[connect_findings.txt](https://github.com/strongloop/loopback-next/files/6755393/connect_findings.txt)

**Comment by raymondfeng** [comment]
Did you try to adjust https://nodejs.org/api/cli.html#cli_uv_threadpool_size_size?

**Comment by vishnu1545** [comment]
Yes, I have tried UV_THREADPOOL_SIZE, this observation logs are with UV_THREADPOOL_SIZE=18 but I have tried with 30 and even higher numbers. But the result is similar, with higher number db inserts are further lined up.

**Comment by saurabhjha092** [comment]
Hi @vishnu1545  @raymondfeng , i am facing a similar issue here trying to insert into postgres  db -having 103 fields . Kind of similar behavior at first loopback-postgres connector not acquiring more than 2 threads and later after certain interval of time - it starts acquiring and relasing threads. In short the Db insertion is too slow [asynchronous nature ] due to which TPS is pathetic .can we bulk insert like more than 1 at a time ?. Here is the link of sample program that i created : https://github.com/saurabhjha092/demo.git [branch: master].
